ptxboa_streamlit.py

```diff
@@ -90,29 +90,52 @@
     colors = pd.read_csv("data/Agora_Industry_Colours.csv")
     st.session_state["colors"] = colors["Hex Code"].to_list()
 
-# calculate results:
-res_costs = pf.calculate_results_list(
-    api, "region", user_data=st.session_state["user_changes_df"]
+# calculate results over different data dimensions:
+costs_per_region = pf.calculate_results_list(
+    api,
+    parameter_to_change="region",
+    parameter_list=None,
+    user_data=st.session_state["user_changes_df"],
+)
+costs_per_scenario = pf.calculate_results_list(
+    api,
+    parameter_to_change="scenario",
+    parameter_list=None,
+    user_data=st.session_state["user_changes_df"],
+)
+costs_per_res_gen = pf.calculate_results_list(
+    api,
+    parameter_to_change="res_gen",
+    parameter_list=[
+        x for x in api.get_dimension("res_gen").index.to_list() if x != "PV tracking"
+    ],
+    user_data=st.session_state["user_changes_df"],
+)
+costs_per_chain = pf.calculate_results_list(
+    api,
+    parameter_to_change="chain",
+    parameter_list=None,
+    user_data=st.session_state["user_changes_df"],
```

Comment:
why do you pass user_data as an argument? The function could access the session state directly

Reply:
changed accordingly